kernel/init.c

```diff
@@ -519,8 +519,12 @@ FUNC_NORETURN void z_cstart(void)
 	z_sys_device_do_config_level(_SYS_INIT_LEVEL_PRE_KERNEL_2);
 
 #ifdef CONFIG_STACK_CANARIES
+#ifdef CONFIG_STACK_CANARIES_WITH_TERMINATOR
+	__stack_chk_guard = z_early_boot_rand32_get() & 0x00ffffffUL;
```

Comment:
I'm not deeply familiar with the literature on this, but wouldn't it be a little better to randomly select which byte to be the terminator, instead of always the MSB?

Reply:
Linux does this way, though it only applies this technique to 64 bits canaries. I'm sending this pr to bring some discussion from other people as well. I'm not 100% sure about the benefits of this, so all comments are welcome.

Reply:
That's another thing, we do have some 64-bit targets, the mask you use will need to be the word length and not a fixed 0xFFFFFF

Reply:
yep, but then we also need to replace rand32_get() with something that returns the word length as well.

Reply:
btw, no objections with the changes, the idea here is make it better

Reply:
rand_get_word()?